https://rt.cpan.org/Public/Bug/Display.html?id=118343

On Wed Oct 12 04:36:31 2016, ppisar wrote:
Hide quoted text

> After upgrading OpenSSL to 1.1.0b, tests fail like this:
> 
> #   Failed test 'use Crypt::SSLeay;'
> #   at t/00-basic.t line 6.
> #     Tried to use 'Crypt::SSLeay'.
> #     Error:  Can't load '/home/test/fedora/perl-Crypt-SSLeay/Crypt-
> SSLeay-0.72/blib/arch/auto/Crypt/SSLeay/SSLeay.so' for module
> Crypt::SSLeay: /home/test/fedora/perl-Crypt-SSLeay/Crypt-SSLeay-
> 0.72/blib/arch/auto/Crypt/SSLeay/SSLeay.so: undefined symbol:
> SSLv2_client_method at /usr/lib64/perl5/DynaLoader.pm line 193.
> #  at t/00-basic.t line 6.
> 
> This is because OpenSSL 1.1.0 removed support for SSLv2
> <https://www.openssl.org/news/cl110.txt>.

Attached patch stops using SSLv3_client_method() with OpenSSL 1.1.1. TheTLS_client_method method is used instead.
Subject: 	Use_TLS_client_method-with-OpenSSL-1.1.1.patch

--- SSLeay.xs.bak	2020-04-30 14:48:03.897259294 +0000
+++ SSLeay.xs	2020-04-30 14:51:14.193744640 +0000
@@ -149,15 +149,15 @@
             ctx = SSL_CTX_new(SSLv23_client_method());
         }
         else if(ssl_version == 3) {
-            ctx = SSL_CTX_new(SSLv3_client_method());
+            ctx = SSL_CTX_new(TLS_client_method());
         }
         else {
-#ifndef OPENSSL_NO_SSL2
+#if !defined OPENSSL_NO_SSL2 && OPENSSL_VERSION_NUMBER < 0x10100000L
             /* v2 is the default */
             ctx = SSL_CTX_new(SSLv2_client_method());
 #else
             /* v3 is the default */
-            ctx = SSL_CTX_new(SSLv3_client_method());
+            ctx = SSL_CTX_new(TLS_client_method());
 #endif
         }
